sdk/src/main/java/com/mapbox/maps/ViewAnnotationManagerImpl.kt

```diff
@@ -143,6 +150,68 @@ internal class ViewAnnotationManagerImpl(
     return renderThread.viewAnnotationMode
   }
 
+  override fun cameraForAnnotations(
+    annotations: List<View>,
+    edgeInsets: EdgeInsets?,
+    bearing: Double?,
+    pitch: Double?
+  ): CameraOptions? {
+    if (mapboxMap.style?.getProjection()?.name == ProjectionName.GLOBE || annotations.isEmpty()) {
+      return null
+    }
+    val viewAnnotationOptions = annotations.mapNotNull {
+      if (it.isVisible) return@mapNotNull getViewAnnotationOptionsByView(it) else null
+    }.filter { it.visible != false }
+    val coordinates = coordinatesFromAnnotations(viewAnnotationOptions)
+    val paddings = calculateEdgeInsets(viewAnnotationOptions, edgeInsets)
+    return mapboxMap.cameraForCoordinates(
+      coordinates,
+      paddings,
+      bearing,
+      pitch
+    )
+  }
+
+  /**
+   * Function to get coordinates from list of [ViewAnnotationOptions].
+   */
+  private fun coordinatesFromAnnotations(annotationOptions: List<ViewAnnotationOptions>): List<Point> {
+    val coordinatesList = mutableListOf<Point>()
+    annotationOptions.forEach {
+      it.geometry?.let { geometry ->
+        coordinatesList.add(Point.fromJson(geometry.toJson()))
+      }
+    }
+    return coordinatesList
+  }
+
+  /**
+   * Calculate paddings to show viewAnnotations.
+   * Get the topMost, leftMost, rightMost, bottomMost annotation options and apply paddings accordingly.
+   */
+  @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+  internal fun calculateEdgeInsets(
+    viewAnnotationOptions: List<ViewAnnotationOptions>,
+    edgeInsets: EdgeInsets? = null
+  ): EdgeInsets {
+    val filteredViewAnnotations = viewAnnotationOptions.filter { it.geometry != null }
+    val topAnnotation =
+      filteredViewAnnotations.maxByOrNull { Point.fromJson(it.geometry!!.toJson()).latitude() }
+    val bottomAnnotation =
+      filteredViewAnnotations.minByOrNull { Point.fromJson(it.geometry!!.toJson()).latitude() }
+    val leftAnnotation =
+      filteredViewAnnotations.minByOrNull { Point.fromJson(it.geometry!!.toJson()).longitude() }
+    val rightAnnotation =
```

Comment:
Do you need to compare lat/lon, or simply offsetX/offsetY should be enough?

Reply:
Ah nevermind, I've confused it with the `ViewAnnotationPositionDescriptor` coordinate, it's offset relative to the geometry point...

Reply:
But at the same time it seems could calculate all annotations in one pass. Now the whole list of annotations is mapped to/from json 4 times.

Reply:
Yeah, i have that in mind, will add a ticket to improve the code after beta.

Reply:
Btw since you anyway expect Point - is Json needed at all? Cant you just cast geometry to point and take lat/lon ? Eg
`(options.geometry as Point).latitude()`?

Reply:
good suggestion. will update that :)